spec/src/language_negotiation_lambda_spec.js

```diff
@@ -0,0 +1,143 @@
+'use strict';
+
+var languageNegotiationLambda = require( '../../src/language_negotiation_lambda' );
+
+describe( 'languageNegotiationLambda', function() {
+    var event = {
+        "Records": [
+            {
+              "cf": {
+                "request": {
+                  "headers": {
+                    "Host": [
+                      "localhost"
+                    ],
+                    "Cookie": [
+                      "SomeCookie=1; AnotherOne=A"
+                    ],
+                    "User-Agent": [
+                      "Test Agent"
+                    ]
+                  },
+                  "clientIp": "2001:cdba::3257:9652",
+                  "httpVersion": "2.0",
+                  "uri": "/",
+                  "method": "GET"
+                },
+                "config": {
+                  "distributionId": "EXAMPLE"
+                }
+              }
+            }
+        ]
+    };
+    var default_header = ['en'];
+
+    beforeEach(function(){
+        delete event.Records[0].cf.request.headers["Accept-Language"];
+        delete event.Records[0].cf.request.headers["X-Accept-Language"];
+    });
+    describe( 'Well formed headers', function(){
+        [
+            { header: "en",                   customHeader: "en" },
+            { header: "es",                   customHeader: "es" },
+            { header: "de",                   customHeader: "en" },
+            { header: "en;q=0.8,es",          customHeader: "es" },
+            { header: "es;q=0.8,en",          customHeader: "en" },
+            { header: "de,en;q=0.8",          customHeader: "en" },
+            { header: "de,es;q=0.8",          customHeader: "es" },
+            { header: "es;q=NaN",             customHeader: "es" },
+            { header: "es;q=NaN,en;q=0.01",   customHeader: "en" },
+            { header: "de,en;q=0.2,es;q=0.8", customHeader: "es" },
+            { header: "EN",                   customHeader: "en" },
+            { header: "ES",                   customHeader: "es" },
+            { header: "EN-US",                customHeader: "en" },
+            { header: "ES-419",               customHeader: "es" }
+        ].forEach( function( headers ) {
+            it( `Resolves header ${headers.header} to ${headers.customHeader}`, function( done ) {
+                event.Records[0].cf.request.headers['Accept-Language'] = [headers.header];
+
+                languageNegotiationLambda.handler( event, { /* context */ }, (err, result) => {
+                    try {
+                        expect( err ).toBeNull();
+                        expect( result.headers['X-Accept-Language'] ).toEqual([headers.customHeader]);
+                        done();
+                    }
+                    catch( error ) {
+                        done( error );
+                    }
+                });
+            });
+        });
+    });
+
+
+    describe( 'malformed headers', function(){
```

Comment:
Can this test be combined with 'Well formed headers'? It runs through the same code.